### PR TITLE
Improve efficiency of a time-based sum of polynomials

### PR DESCRIPTION
**Pull request by jrising**
This replaces a SumCurve of PolynomialCurves, each given a single selected time, with a single SumbyTimePolynomialCurve which takes a K x T matrix of coefficients and applies them to a vector of weather values.

### CODE REVIEW

**Comment by jrising** [comment]
@brews Just a new `smart_curve` class for supporting my sum-by-time efficiency improvements. Code review?

File: openest/generate/smart_curve.py [hunk 1]
```diff
@@ -179,6 +179,80 @@ def format(self, lang):
...
+                    repterms.append(r"%s_1 \cdot %s(%s)^%d" % (coeffvar, funcvar, variable, ii + 1))
+            elif lang == 'julia':
+                if isinstance(self.variables[0], str):
+                    repterms.append(r"sum(%s[1,:] * %s^%d)" % (coeffvar, variable, ii + 1))
+                else:
+                    funcvar = formatting.get_function()
+                    funcvars[self.variables[ii]] = funcvar
+                    repterms.append(r"sum(%s[1,:] * %s(%s)^%d)" % (coeffvar, funcvar, variable, ii + 1))
+
+        result = {'main': FormatElement(' + '.join(repterms))}
```
Comment (brews) [inline openest/generate/smart_curve.py:249]: Ignore me if I'm wrong but I don't think we import `FormatElement`, so unresolved reference here.

File: openest/generate/smart_curve.py [hunk 2]
```diff
@@ -179,6 +179,80 @@ def format(self, lang):
...
+        for variable in funcvars:
+            result[funcvars[variable]] = FormatElement(self.descriptions.get(variable, "Unknown"))
```
Comment (brews) [inline openest/generate/smart_curve.py:251]: Like previous comment, I think `FormatElement` is unresolved reference here.

**Comment by jrising** [comment]
@brews That does appear to be the case. Anyway, I've now referenced them. Thanks!

**Pull request merged**